server/src/main/java/org/opensearch/extensions/rest/RestInitializeExtensionAction.java

```diff
@@ -62,36 +70,73 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
-                parser.nextToken();
-                if ("name".equals(currentFieldName)) {
-                    name = parser.text();
-                } else if ("uniqueId".equals(currentFieldName)) {
-                    uniqueId = parser.text();
-                } else if ("hostAddress".equals(currentFieldName)) {
-                    hostAddress = parser.text();
-                } else if ("port".equals(currentFieldName)) {
-                    port = parser.text();
-                } else if ("version".equals(currentFieldName)) {
-                    version = parser.text();
-                } else if ("opensearchVersion".equals(currentFieldName)) {
-                    openSearchVersion = parser.text();
-                } else if ("minimumCompatibleVersion".equals(currentFieldName)) {
-                    minimumCompatibleVersion = parser.text();
-                } else if ("dependencies".equals(currentFieldName)) {
-                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
-                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
-                        dependencies.add(ExtensionDependency.parse(parser));
-                    }
+        Tuple<? extends MediaType, Map<String, Object>> unreadExtensionTuple = XContentHelper.convertToMap(
+            request.content(),
+            false,
+            request.getXContentType().xContent().mediaType()
+        );
+        Map<String, Object> extensionMap = unreadExtensionTuple.v2();
+
+        ExtensionScopedSettings extAdditionalSettings = new ExtensionScopedSettings(extensionsManager.getAdditionalSettings());
+
+        try {
+            // checking to see whether any required fields are missing from extension.yml file or not
+            String[] requiredFields = {
+                "name",
+                "uniqueId",
+                "hostAddress",
+                "port",
+                "version",
+                "opensearchVersion",
+                "minimumCompatibleVersion" };
+            List<String> missingFields = Arrays.stream(requiredFields)
+                .filter(field -> !extensionMap.containsKey(field))
+                .collect(Collectors.toList());
+            if (!missingFields.isEmpty()) {
+                throw new IOException("Extension is missing these required fields : " + missingFields);
+            }
+
+            // Parse extension dependencies
+            List<ExtensionDependency> extensionDependencyList = new ArrayList<ExtensionDependency>();
+            if (extensionMap.get("dependencies") != null) {
+                List<HashMap<String, ?>> extensionDependencies = new ArrayList<>(
+                    (Collection<HashMap<String, ?>>) extensionMap.get("dependencies")
+                );
+                for (HashMap<String, ?> dependency : extensionDependencies) {
+                    extensionDependencyList.add(
+                        new ExtensionDependency(
```

Comment:
Before creating the object for `ExtensionDependency`. Do you think we need a check for `uniqueId` and `version` like [this](https://github.com/opensearch-project/OpenSearch/pull/8414/files#diff-97fa31f016b85d4289975172a977d19d935284b729c3611bf63934ca375077ccL88).

Reply:
I added the checks in now. This area is not currently covered by tests and the tests should be expanded to cover more scenarios like required values missing.